x-pack/plugins/observability_solution/asset_manager/server/lib/auth/privileges.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  IScopedClusterClient,
+} from '@kbn/core/server';
+import { ENTITY_BASE_PREFIX } from "@kbn/assetManager-plugin/common/constants_entities";
+
+export const requiredRunTimePrivileges = {
+  // all of
+  index: [
+    {
+      names: [`${ENTITY_BASE_PREFIX}*`],
+      privileges: [
+        'create_doc',
+        'auto_configure',
+        'read',
+      ],
+    },
+    {
+      // how do we know ahead of time what we need to include here??
```

Comment:
i wonder if we need to dynamically read the index patterns defined in all the built in entity definitions.

the only problem there is that an existing valid API key could become invalid if new built-in definitions are added which require read permissions on additional indices. we don't want that to happen. 

perhaps we define a fixed set of indices which can be accessed by the built-in definitions, which matches the set of indices defined here, so that everything is aligned without having to do any runtime checks.

Reply:
I feel like this is where the Admin who setup the cluster needs to decide what permissions to grant to the key. We can document what indices we query OTB but they are responsible for how those are restricted.

Reply:
> the only problem there is that an existing valid API key could become invalid if new built-in definitions are added which require read permissions on additional indices. we don't want that to happen.

This would happen when the user/admin upgraded the to a new version? We'd need to document this or come up with a way to notify the admin that they need to re-enable the entity discovery. Maybe we could have a "reset" API that would allow them to just click reset (or call the API) and we take care of the rest.

Reply:
What if we added a version to the metadata of the API key, then change `PUT /api/entities/enablement` to do the following:

1. Check if the API Key is the current version, if it is, we just `noop` and return the current behavior (all good)
2. If the API Key is an older version, we delete the API Key and create the new one with the new permissions
3. Update or install the definitions (out of scope for this PR)

This way SREs can just call the PUT request as part of setting up a new Kibana instance OR any time make an update to the Teraform definition.  It can be called as many times as they want but the only time it does something:

- Fresh install
- Upgrade to new version of the stack

Reply:
we discussed this and will take a best guess approach by hardcoding a preset of patterns that builtin definitions are allowed to use